sql/core/src/main/java/org/apache/spark/sql/execution/UnsafeKVExternalSorter.java

```diff
@@ -89,7 +89,7 @@ public UnsafeKVExternalSorter(StructType keySchema, StructType valueSchema,
         taskMemoryManager, recordComparator, prefixComparator, Math.max(1, map.numElements()));
 
       final int numKeyFields = keySchema.size();
-      BytesToBytesMap.BytesToBytesMapIterator iter = map.iterator();
+      BytesToBytesMap.BytesToBytesMapIterator iter = map.destructiveIterator();
```

Comment:
i don't think you can call the destructive iterator here, since we are using the page to do sorting in-place.